### PR TITLE
Fix subgroups leading catch all matcher

### PR DESCRIPTION
**Pull request by tisonkun**
This refers to #1401.

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @nedtwigg @murdos

File: testlib/src/test/java/com/diffplug/spotless/java/ImportOrderStepTest.java [hunk 1]
```diff
@@ -41,6 +41,12 @@ void sortImportsFromArrayWithSubgroups() {
 		StepHarness.forStep(step).testResource("java/importsorter/JavaCodeUnsortedImportsSubgroups.test", "java/importsorter/JavaCodeSortedImportsSubgroups.test");
 	}
 
+	@Test
+	void sortImportsFromArrayWithSubgroupsLeadingCatchAll() {
+		FormatterStep step = ImportOrderStep.forJava().createFrom("\\#|");
```
Comment (tisonkun) [inline testlib/src/test/java/com/diffplug/spotless/java/ImportOrderStepTest.java:46]: Verified that it fails without this change.

**Comment by nedtwigg** [comment]
Thanks for the fix! I'm trying to release today, but migrating our deployment from CircleCI to GitHub has been surprisingly tricky...

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.13.0` and `plugin-maven 2.30.0`.

**Comment by tisonkun** [comment]
@nedtwigg Thank you! I'm trying this version now :)

**Pull request merged**